symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -63,8 +64,96 @@ void test_cwrapper() {
     basic_str_free(s);
 }
 
+void test_CVectorInt1()
+{
+    // Allocate on heap
+    CVectorInt *vec = vectorint_new();
+    vectorint_push_back(vec, 5);
+    assert(vectorint_get(vec, 0) == 5);
+    vectorint_free(vec);
+}
+
+struct X {
+    void *x;
+};
+
+void test_CVectorInt2()
+{
+    // Allocate on stack
+    CVectorInt *vec;
+
+    char data1[1];  // Not aligned properly
+    vec = (CVectorInt*)data1;
+    assert(vectorint_placement_new(vec, sizeof(data1)) == 2);
+
+    struct X data2[1];  // Aligned properly but small
+    vec = (CVectorInt*)data2;
+    assert(vectorint_placement_new(vec, sizeof(data2)) == 1);
+
+    char data3[50]; // Aligned properly and enough size to fit std::vector<int>
```

Comment:
We check it at runtime. It seems it is aligned properly in Debug mode, but
not in Release mode (after I enabled the assert statement). I need to look
into it more.

Sent from my mobile phone.
On Jul 4, 2015 3:13 AM, "Isuru Fernando" <notifications@github.com> wrote:

> In symengine/tests/cwrapper/test_cwrapper.c
> <https://github.com/sympy/symengine/pull/495#discussion_r33886143>:
>
> > +};
> > +
> > +void test_CVectorInt2()
> > +{
> > +    // Allocate on stack
> > +    CVectorInt *vec;
> > +
> > +    char data1[1];  // Not aligned properly
> > +    vec = (CVectorInt*)data1;
> > +    assert(vectorint_placement_new(vec, sizeof(data1)) == 2);
> > +
> > +    struct X data2[1];  // Aligned properly but small
> > +    vec = (CVectorInt*)data2;
> > +    assert(vectorint_placement_new(vec, sizeof(data2)) == 1);
> > +
> > +    char data3[50]; // Aligned properly and enough size to fit std::vector<int>
>
> Is this aligned properly?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/symengine/pull/495/files#r33886143>.
>